cfme/fixtures/utility_vm.py

```diff
@@ -32,29 +32,40 @@
 from cfme.utils.wait import TimedOutError
 
 
+# I know this isn't good for reuse, I am just trying to get something to work.
 def _trying_ips(vm, attempts=60, interval=10):
     for attempt in range(attempts):
         for ip in getattr(vm, 'all_ips', []):
-            yield ip
-            return
+            try:
+                if ip.split('.')[0] == '10':
+                    yield ip
+                    return
+            except AttributeError:
+                pass
         time.sleep(interval)
 
 
 @contextmanager
 def connect_ssh(vm, creds):
     for ip in _trying_ips(vm):
-        try:
-            with SSHClient(hostname=ip, username=creds.principal, password=creds.secret) as client:
-                logger.info("SSH connected to IP %s", ip)
-                result = client.run_command("true")
-                if not result.success:
-                    raise Exception(f"Command `true` failed on ip {ip}.")
-                yield client
-                return
-        except Exception as ex:
-            logger.info("Failed to connect with SSH to %s: %s", ip, ex)
+        for ssh_attempt in range(55):  # I know ... hardcoded. I'll fix
+            try:
+                with SSHClient(
+                        hostname=ip,
+                        username=creds.principal,
+                        password=creds.secret
+                ) as client:
+                    logger.info("SSH connected to IP %s", ip)
+                    result = client.run_command("true")
+                    if not result.success:
+                        raise Exception(f"Command `true` failed on ip {ip}.")
+                    yield client
+                    return
+            except Exception as ex:
+                logger.info("Failed to connect with SSH to %s: %s", ip, ex)
+                time.sleep(20)
```

Comment:
_trying_ips is sleeping after trying all the different IPs returned from the mgmt system. I wonder why do you want it to sleep after every unsuccessful connection.

Reply:
To wait for the SSH client? We keep going around in circles about this. Am I missing something? We wait until we get IP,  then we need to wait for the SSH client to become active. I have been debugging and AFAICT once, we get an IP, we only try to get an SSH connection once. Am I wrong? I am running this code locally and I am unable to get it to pass. SSH is timing out. Maybe the difference between running locally and running in PRT is causing some of this confusion. I'd really like to be able to run this test in both places.